### PR TITLE
Add TileJSON

### PR DESCRIPTION
**Pull request by lossyrob**
This PR adds the ability to specify web map links via [TileJSON](https://github.com/mapbox/tilejson-spec) through specifying an Asset with `tiles` in the Asset roles.

### CODE REVIEW

**Comment by lossyrob** [comment]
@m-mohr wasn't sure how to validate this in the json schema; if you have any pointers I can work on getting it validated if possible.

**Comment by m-mohr** [comment]
@lossyrob Yeah, adding in assets makes the whole validation much more complicated. So I guess there's a reason why this is in assets instead of links? Would it also work in links? Are people meant to download this? I won't have the time to update the JSON Scherma accordingly any time soon, unfortunately.

**Comment by lossyrob** [comment]
@m-mohr it feels more like an asset then a link - similar to metadata files that are associated e.g. with scenes. People wouldn't necessarily download it, but front-ends would pull the JSON data in and feed it to a map layer e.g. https://github.com/microsoft/planetary-computer-apis/blob/main/pctiler/pctiler/endpoints/templates/item_preview.html#L20

**Comment by m-mohr** [comment]
@lossyrob Yes, I agrre, but to simplify the schema and extension a bit, do you think it makes sense to split into a web-map-links and a tile-json extension or so? Otherwise, we shouldn't merge until a JSON Schema can be provided.

**Comment by cholmes** [comment]
To me it actually makes *slightly* more sense in 'links' than assets. I can see the argument for asset, since it's a concrete json metadata file. But I feel like it serves the same purpose as like a WMTS. It's more the information a client needs to get an alternate access to the data, versus the data itself, or its metadata. Obviously it's a fuzzy line, since one could argue any service description is 'metadata'. XYZ seems the one that most clearly should be a 'link', since there's no metadata. WMTS feels since the capabilities document is a form of 'metadata'. 

The next one to come will be OGC API - Tiles JSON, which is closer to a TileJSON, and that one feels to me like it should be a link. I feel like the assets should be a bundle that you'd take offline and everything would still work, while links can be services. So like a zipped set of tiles would make sense to me as an asset, but a TileJSON seems more like the description of the online service.

I know that Planetary Computer has it as an asset now, and I can totally see why that would make sense if it was the only one. And I know it can be hard to change, and I don't feel that strongly that it must be a link.

**Comment by m-mohr** [comment]
To keep the extension lightweight, I'd also prefer having it in links and agree with Chris. Otherwise, I'd probably prefer to split "tilejson as asset" into a separate extension.

@lossyrob Would it work for you to add them additionally as links to MS PC if we add them as links here?

**Comment by lossyrob** [comment]
If the best practice we're moving forward with is "any description of an online service about the data is best represented as a link rather than an asset", that seems reasonable to me.

I can modify the pc to add it as a link and keep the asset around for a while in case anyone was using them.

I'll update this PR with the move from asset to link when I get a chance - would this hold up any release plans?

**Comment by m-mohr** [comment]
👍 My plan was to release "soon" ;-) Let's see whether you had the chance before there's a need for a release.

File: README.md [hunk 1]
```diff
@@ -31,22 +31,34 @@ in the top-level object of the document via the `attribution` field as defined i
...
-| rel             | string               | **REQUIRED**. Must be set to `wmts`. |
-| href            | string               | **REQUIRED**. Link to the WMTS, without any WMTS specific query parameters. |
-| wmts:layer      | string\|\[string]    | **REQUIRED**. The layers to show on the map, either a list of layer names or a single layer name. |
+| Field Name      | Type                 | Description                                                                                                      |
+| --------------- | -------------------- | ---------------------------------------------------------------------------------------------------------------- |
+| rel             | string               | **REQUIRED**. Must be set to `wmts`.                                                                             |
+| href            | string               | **REQUIRED**. Link to the WMTS, without any WMTS specific query parameters.                                      |
+| wmts:layer      | string\|\[string]    | **REQUIRED**. The layers to show on the map, either a list of layer names or a single layer name.                |
```
Comment (emmanuelmathot) [inline README.md]: why is this required? layer can be enclosed in the link.

**Review by emmanuelmathot — APPROVED** [review]
I like the `link` level since it allows to specify in a generic way an "offering" to the asset/item

File: README.md (hunk 1 above)
Comment (m-mohr) [reply README.md]: Because the link doesn't contain parameters.

**Comment by m-mohr** [comment]
@lossyrob Is this still of interest? I'd like to finally release this and it's ready to be released, but the changes to links in this PR are still missing. It also seems like the TileJSON in MS PC is still in assets.

Edit: I'll just release 1.0.0 and we can then easily add TileJSON in a 1.1.0.

**Comment by lossyrob** [comment]
@m-mohr just updated (edit: fixing CI)

**Comment by m-mohr** [comment]
Thanks, will be re-released as 1.0.0.
